### PR TITLE
Add file set authorization routes

### PR DESCRIPTION
**Pull request by kdid**
- `/file-sets/{id}/authorization` returns `204` if file set image is authorized according to our logic and `403` if not

### CODE REVIEW

File: src/handlers/get-file-set-auth.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    body: body,
+  });
+}
+
+function isAllowedVisibility(token, visibility, readingRoom) {
+  switch (visibility) {
+    case "Public":
+      return true;
+    case "Institution":
+      isValidToken(token);
```
Comment (mbklein) [inline src/handlers/get-file-set-auth.js:53]: Missing the `return` keyword.
```suggestion
      return isValidToken(token);
```

File: template.yaml [hunk 2]
```diff
@@ -190,6 +190,26 @@ Resources:
             ApiId: !Ref dcApi
             Path: /file-sets/{id}
             Method: GET
+  getFileSetAuthFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: handlers/get-file-set-auth.handler
+      Description: Authorizes access to a file set.
```
Comment (mbklein) [inline template.yaml:197]: `API_TOKEN_SECRET` isn't part of the default set of environment variable every function gets, so in order to decode the user token, this function needs
```suggestion
      Description: Authorizes access to a file set.
      Environment:
        Variables:
          API_TOKEN_SECRET: !Ref ApiSecret
```

**Review by mbklein — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
   "author": "nulib",
   "license": "Apache-2.0",
   "dependencies": {
-    "dc-api": "file:./src"
+    "dc-api": "file:./src",
+    "lodash.isobject": "^3.0.2"
```
[the remark this answers is not in the document]
Comment (kdid) [reply package.json:10]: Right. Totally forgot. Will fix

File: src/handlers/get-file-set-auth.js (hunk 1 above)
Comment (kdid) [reply src/handlers/get-file-set-auth.js:53]: Always forget the return in JS

Comment (mbklein) [reply src/handlers/get-file-set-auth.js:53]: Same. Ruby and Elixir have ruined us. 😄

**Review by mbklein — APPROVED** [review]

**Pull request merged**